app/components/navigation/navigation-item.tsx

```diff
@@ -0,0 +1,10 @@
+import { useState } from "react";
+import INavigationItem from '@/app/models/INavigationItem';
+export default function NavigationItem({ inputAttr, labelAttr, listAttr, children, hasSubpages }: INavigationItem) {
+    const [isHidden, setIsHidden] = useState(true);
+    return (<>
+        <input id={inputAttr.id} className={inputAttr.classes} type="checkbox" onChange={(e)=> setIsHidden(!e.target.value)}/>
+        <label htmlFor={inputAttr.id} id={labelAttr.id} aria-controls={labelAttr.aria.controls} data-testid={labelAttr.testId} className={labelAttr.classes }>{labelAttr.content}</label>
+        {hasSubpages && <ul className={listAttr.classes}  id={listAttr.id} data-testid={listAttr.testId} aria-hidden={isHidden} hidden={isHidden}>{children}</ul>}
+    </>)
+}
```

Comment:
nit: do we not expect newlines EOF yet 😅

Reply:
Haha fair. I'm used to this being automatically added, so I'm still adjusting to manually adding it.